internal/collector/deployment.go

```diff
@@ -20,7 +21,11 @@ func (f *Factory) ReconcileDeployment(k8sClient client.Client, namespace string,
 	tlsProfile, _ := tls.FetchAPIServerTlsProfile(k8sClient)
 	desired := f.NewDeployment(namespace, f.ResourceNames.DaemonSetName(), trustedCABundle, tls.GetClusterTLSProfileSpec(tlsProfile))
 	utils.AddOwnerRefToObject(desired, owner)
-	return reconcile.Deployment(k8sClient, desired)
+	return reconcile.Deployment(k8sClient, desired, func(current, desired apps.Deployment) (bool, string) {
+		currentHash, _ := current.Spec.Template.Annotations[annotationConfigHash]
+		desiredHash, _ := desired.Spec.Template.Annotations[annotationConfigHash]
+		return currentHash == desiredHash, ""
```

Comment:
Why not add this to `deployments.AreSame` instead of having an additional point where comparisons are made?